pkg/agent/ocp.go

```diff
@@ -68,16 +68,14 @@ func (ocp *ClusterOpenShiftAPIClient) AreClusterOperatorsInitialized() (bool, er
 
 	failing := configv1.ClusterStatusConditionType("Failing")
 
-	operators, err := ocp.ConfigClient.ConfigV1().ClusterOperators().List(ocp.ctx, metav1.ListOptions{})
+	version, err := ocp.ConfigClient.ConfigV1().ClusterVersions().Get(ocp.ctx, "version", metav1.GetOptions{})
 	if err != nil {
-		return false, errors.Wrap(err, "Listing ClusterOperator objects")
+		return false, errors.Wrap(err, "Getting ClusterVersion object")
 	}
 
-	operator := operators.Items[0]
-
-	if cov1helpers.IsStatusConditionTrue(operator.Status.Conditions, configv1.OperatorAvailable) &&
-		cov1helpers.IsStatusConditionFalse(operator.Status.Conditions, failing) &&
-		cov1helpers.IsStatusConditionFalse(operator.Status.Conditions, configv1.OperatorProgressing) {
+	if cov1helpers.IsStatusConditionTrue(version.Status.Conditions, configv1.OperatorAvailable) &&
```

Comment:
The install-complete returns success right after the first two nodes form the control plane. It should wait for all three nodes to join.

I found the status check on Available is not working as expected. It reports as available here, but "oc get clusterversion" shows False for Available.

```
[rwsu@hardprov-fx2-22 ostest]$ ./openshift-install agent wait-for install-complete
INFO Cluster Kube API Initialized                 
INFO Bootstrap configMap status is complete       
INFO cluster bootstrap is complete                
INFO Cluster is installed                         
INFO Install complete!                            
INFO To access the cluster as the system:admin user when using 'oc', run 
INFO     export KUBECONFIG=/home/rwsu/go/src/github.com/openshift-metal3/dev-scripts/ocp/ostest/auth/kubeconfig 
INFO Access the OpenShift web-console here: https://console-openshift-console.apps.ostest.test.metalkube.org 
[rwsu@hardprov-fx2-22 ostest]$ export KUBECONFIG=./auth/kubeconfig 
[rwsu@hardprov-fx2-22 ostest]$ oc get clusterversion
NAME      VERSION   AVAILABLE   PROGRESSING   SINCE   STATUS
version             False       True          22m     Working towards 4.11.0-0.nightly-2022-07-29-173905: 796 of 802 done (99% complete)
```

Reply:
This is resolved with the patch to check the cluster operators added back in. :)